src/Compilers/CSharp/Test/Emit2/Attributes/AttributeTests.cs

```diff
@@ -5043,6 +5043,228 @@ class C<T>
             });
         }
 
+        [Fact]
+        public void TestMainAttributes()
+        {
+            var source = CreateCompilation(@"
+using System;
+
+[main: My(""one"")]
+Console.WriteLine(""Hello World"");
+
+public class MyAttribute : Attribute { public MyAttribute(string name) {} }
+");
+
+            var mainAttrs = source.GetEntryPoint(default).GetAttributes();
+
+            var attribute = Assert.Single(mainAttrs);
+            attribute.VerifyValue(0, TypedConstantKind.Primitive, "one");
+            Assert.Equal(@"MyAttribute(""one"")", attribute.ToString());
+
+            CompileAndVerify(source, expectedOutput: "Hello World");
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void TestMainAttributes_IL()
+        {
+            var source = CreateCompilation(@"
+using System;
+
+[main: My(""one"")]
+Console.WriteLine(""Hello World"");
+
+public class MyAttribute : Attribute { public MyAttribute(string name) {} }
+");
+
+            CompileAndVerify(source, expectedOutput: "Hello World")
+                .VerifyTypeIL("Program", @"
```

Comment:
>VerifyTypeIL [](http://example.com/codeflow?start=17&length=12)

I think it is an overkill to use IL to confirm presense of the attribute in metadata. IL includes to much of unrelated information and is hard to verify. I would use `symbolVerifier` to locate the method symbol and then check ```GetAttributes()``` for it.